packages/mui-material/src/Switch/Switch.js

```diff
@@ -140,33 +148,38 @@ const SwitchSwitchBase = styled(SwitchBase, {
         backgroundColor: 'transparent',
       },
     },
-    ...(ownerState.color !== 'default' && {
-      [`&.${switchClasses.checked}`]: {
-        color: (theme.vars || theme).palette[ownerState.color].main,
-        '&:hover': {
-          backgroundColor: theme.vars
-            ? `rgba(${theme.vars.palette[ownerState.color].mainChannel} / ${
-                theme.vars.palette.action.hoverOpacity
-              })`
-            : alpha(theme.palette[ownerState.color].main, theme.palette.action.hoverOpacity),
-          '@media (hover: none)': {
-            backgroundColor: 'transparent',
+    variants: [
+      ...Object.entries(theme.palette)
+        .filter(([, value]) => value.main && value.light) // check all the used fields in the style below
```

Comment:
This line broke our production code. We have an entry in the theme object that is sometimes undefined. This code doesn't handle that and assumes all values should be an object (which they are typically), but can also be numbers and strings. It doesn't fail there, but when the value is `undefined`.

```suggestion
        .filter(([, value]) => value?.main && value?.light) // check all the used fields in the style below
```

Reply:
Could you create a new issue so that I can assign to myself?